Wikipedia/Code/SavedArticlesFetcher.swift

```diff
@@ -255,14 +261,54 @@ private extension SavedArticlesFetcher {
     
     func didFailToFetchArticle(with key: String, error: Error) {
         operateOnArticles(with: key) { (article) in
-            article.updatePropertiesForError(error as NSError)
-            article.isDownloaded = !isErrorRecoverableOnRetry(error)
+            handleFailure(with: article, error: error)
         }
     }
     
-    func isErrorRecoverableOnRetry(_ error: Error) -> Bool {
-        // TODO: handle different errors differently - set isDownloaded = true on items that need to be skipped (errors that won't recover on retry)
-        return true
+    func handleFailure(with article: WMFArticle, error: Error) {
+        var underlyingError: Error = error
+        if let writerError = error as? ArticleCacheDBWriterError {
```

Comment:
@joewalsh note there's also CacheFileWriter that could spit out errors too, if you wanted to save those specific errors. I think now it would probably default to .apiFailed if it's not specifically an NSFileWriteOutOfSpaceError, which could be enough.

Reply:
@tonisevener thanks for the heads up! Seems like `.apiFailed` is enough for now but might be nice to split out in the future